src/ProjNavigator/tcl_command_integration.cpp

```diff
@@ -44,6 +44,7 @@ bool TclCommandIntegration::TclSetTopModule(int argc, const char *argv[],
   const QString topModule = QString(argv[1]);
   m_projManager->setCurrentFileSet(m_projManager->getDesignActiveFileSet());
   int ret = m_projManager->setTopModule(topModule);
+  if (argc == 4) m_projManager->setTopModuleLibrary(QString(argv[3]));
```

Comment:
Should this only fire when exactly 4 args are provided or is argc > 4 a possible valid scenario as well?

Reply:
Similar question. Do we have a argument parser to consume as needed instead of hard-coding numbers here ?